### PR TITLE
55 re render projects and tasks after a postpatch

### PR DESCRIPTION
**Pull request by loganpaulmatheny**
### Summary: This PR will fix all the POST / PATCH re-rendering problems!
- remover user from the dependency array
- set GS within the respective api calls

### Type of change

- [ ]  New feature 
- [x]  Bug fix 
- [ ]  Refactor
- [ ]  Testing
- [ ]  Comment: 

### Checklist:

- [x]  Code needs to be tested
- [ ]  All tests are passing
- [x]  Create new branch AFTER you merge
- [x]  Whats next: Edit 'editModal' to allow users to change the status of tasks and update the UI so that canvas and file upload match/are integrated in a way that makes sense.

### CODE REVIEW

**Comment by joh-ann** [comment]
LGTM, thanks for cleaning my messy code up! I would like to discuss further about us running a GET and updating the GS after our POST/PATCH requests

File: src/apiCalls.js [hunk 1]
```diff
@@ -15,80 +17,114 @@ export const fetchData = () => {
...
-  });
+  )
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error(`Failed to POST`);
+      }
+      fetchData().then((data) => {
+        dispatch(setUserData(data.data));
+      });
+    })
```
Comment (epintozzi) [inline src/apiCalls.js:51]: It looks like most of this PR is some stylistic cleanup, but I noticed these lines added to a few places. Can you share what this is doing and why you added it?

Comment (loganpaulmatheny) [reply src/apiCalls.js:51]: This line is making the GET request following a POST or PATCH request in order to GET the updated user data and set the global state (redux) to the appropriate updated state via dispatching the reducer called setUserData.

Comment (epintozzi) [reply src/apiCalls.js:51]: Neat! Thanks for the info.

**Pull request merged**